### PR TITLE
Minor refactor

### PR DESCRIPTION
**Pull request by fisker**
## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by sosukesuzuki — APPROVED** [review]

File: src/language-js/printer-estree.js [hunk 1]
```diff
@@ -811,7 +799,7 @@ function printPathNoParens(path, options, print, args) {
 
     default:
       /* istanbul ignore next */
-      throw new Error("unknown type: " + JSON.stringify(node.type));
+      throw Object.assign(new Error("Unknown node type"), { node });
```
Comment (thorn0) [inline src/language-js/printer-estree.js:802]: Having `node.type` in the message is convenient. Let's keep it.

**Comment by thorn0** [comment]
While you're at these minor cleanup, I noticed the same function `isTypeCastComment` is included twice in two different files. I guess it's not intended.

**Comment by thorn0** [comment]
BTW, there is a thing I've wanted to ask for a while. Why do we use this style:
```js
function foo() { ... }
export { foo }
```
instead of
```js
export function foo() { ... }
```
?

**Comment by fisker** [comment]
> Why do we use this style

Just personal preference, easier to see what module exported.

**Review by thorn0 — APPROVED** [review]

**Comment by thorn0** [comment]
`isTypeCastComment` has been left in the exports :)

**Comment by fisker** [comment]
Stupid me.

**Pull request merged**